graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/export/ExportClient.java

```diff
@@ -39,7 +41,19 @@ public ExportClient(ElasticsearchClient client) {
 
     public SearchResponse search(SearchRequest request, String errorMessage) {
         try {
-            return this.client.search(request, errorMessage);
+            final SearchResponse response = this.client.search(request, errorMessage);
+            if (response.getFailedShards() > 0) {
+                final List<Throwable> shardFailures = Arrays.stream(response.getShardFailures())
+                        .map(ShardOperationFailedException::getCause)
+                        .toList();
```

Comment:
Creation of this list seems unnecessary. You could collect the errors with longer chain of stream operations in line 49.